### DISCUSSION TITLE
Dictation during pressing key

### DISCUSSION THREAD

**Discussion by hi4nik** [EpicenterHQ/epicenter#307] [Ideas]
Please add a feature of dictation during pressing specific key.  I mean, when I press hotkey, dictation is starting and continue during I release the key.

**Comment by doxgt** [comment]
See https://github.com/braden-w/whispering/issues/181.

In the meantime, what you can do is use platform-specific tools to "send" Whispering's recording toggle and create a secondary push-to-talk routine.  

On Windows, you could use AutoHotkey (see [KeyWait](https://www.autohotkey.com/docs/v2/lib/KeyWait.htm)) or AutoIt, among others.  I know next-to-nothing about macOS.  But I wonder if you may be able to use Apple Script or similar.  I'm fairly certain on Linux you can access kindred-spirited tools for like purposes.

**Comment by hi4nik** [comment]
Yes, it works, thank you for advice !

**Reply by doxgt** [reply]
While you are at it, if you are on Windows and if it is AutoHotkey that you are going to harness, then take a look at its [Sound Functions](https://www.autohotkey.com/docs/v2/lib/Sound.htm). 

I dare to reckon it will completely solve the other issue of yours: https://github.com/braden-w/whispering/discussions/306

Just before you send the Whispering recording toggle keyboard shortcut, you can mute whatever you need to mute.  A short delay after KeyWait, then unmute.

Simple as that.

Cheers!

**Comment by braden-w** [comment]
Now resolved in the latest release, since #489 !
